luatest/server.lua

```diff
@@ -550,6 +670,18 @@ function Server:connect_net_box()
         error(connection.error)
     end
     self.net_box = connection
+
+    if self.config_file ~= nil then
+        -- Replace the ready condition.
+        local saved_eval = self.net_box.eval
+        self.net_box.eval = function(self, expr, args, opts) -- luacheck:ignore 432
+            if expr == 'return _G.ready' then
+                expr = "return require('config'):info().status == 'ready' or " ..
+                          "require('config'):info().status == 'check_warnings'"
+            end
+            return saved_eval(self, expr, args, opts)
+        end
+    end
```

Comment:
This hacky way to replace the ready condition was used in the external wrapper. However, we don't need it if we can modify the `wait_until_ready` method. I propose to check `self.config_file` right in `wait_until_ready` and choose between `_G.ready` and `config:info()` checks.

Reply:
Fixed.